File: MotorcycleRideLog/src/MotorcycleRideLog.WebApp/projects/motorcycle-ride-log/src/app/models/maintenance.ts
import { MaintenanceType } from './maintenance-type';

export interface Maintenance {
  maintenanceId: string;
  userId: string;
  motorcycleId: string;
  maintenanceDate: string;
  type: MaintenanceType;
  mileageAtMaintenance?: number;
  description: string;
  cost?: number;
  serviceProvider?: string;
  partsReplaced?: string;
  notes?: string;
  createdAt: string;
}

export interface CreateMaintenance {
  userId: string;
  motorcycleId: string;
  maintenanceDate: string;
  type: MaintenanceType;
  mileageAtMaintenance?: number;
  description: string;
  cost?: number;
  serviceProvider?: string;
  partsReplaced?: string;
  notes?: string;
}

export interface UpdateMaintenance {
  maintenanceId: string;
  motorcycleId: string;
  maintenanceDate: string;
  type: MaintenanceType;
  mileageAtMaintenance?: number;
  description: string;
  cost?: number;
  serviceProvider?: string;
  partsReplaced?: string;
  notes?: string;
}

File: MotorcycleRideLog/src/MotorcycleRideLog.Core/Model/MaintenanceAggregate/Maintenance.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace MotorcycleRideLog.Core;

public class Maintenance
{
    public Guid MaintenanceId { get; set; }
    public Guid UserId { get; set; }
    public Guid MotorcycleId { get; set; }
    public DateTime MaintenanceDate { get; set; }
    public MaintenanceType Type { get; set; }
    public int? MileageAtMaintenance { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal? Cost { get; set; }
    public string? ServiceProvider { get; set; }
    public string? PartsReplaced { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; }
}
